src/Storage/Repository/InstanceRepository.cs

```diff
@@ -553,11 +552,13 @@ public async Task<Instance> GetOne(int instanceOwnerPartyId, Guid instanceGuid)
         /// <inheritdoc/>
         public async Task<Instance> Update(Instance item)
         {
+            List<DataElement> dataElements = item.Data;
             PreProcess(item);
 
             Instance instance = await Container.UpsertItemAsync<Instance>(item, new PartitionKey(item.InstanceOwner.PartyId));
 
-            await PostProcess(instance);
+            instance.Data = dataElements;
```

Comment:
This is probably fine to do. I initially feared it would come back and bite us one day, but I can't see how that could happen at this point in time. The caller shouldn't expect changes to data elements when updating the instance.

Reply:
Only case I can imagine in the future is if we were to start using server side triggers do more of the cross-document status synchronization (lastChanged timestamp for example). But that would require a larger refactoring to move dataElements into the same collection as instances, since server side triggers do not yet support cross-collection operations.